### PR TITLE
Dynamic Yield: turning Destination into Audience Destination

### PR DESCRIPTION
**Pull request by joe-ayoub-segment**
Turning DY Destination into an Audience Destination

## Testing

None done yet. Do not deploy

### CODE REVIEW

File: packages/destination-actions/src/destinations/dynamic-yield-audiences/syncAudience/index.ts [hunk 1]
```diff
@@ -42,57 +66,90 @@ const action: ActionDefinition<Settings, Payload> = {
...
-    const audienceID = payload.segment_audience_id
-    const audienceValue = d?.rawData?.properties?.[audienceName] ?? d?.rawData?.traits?.[audienceName]
+    const { audienceSettings, payload, settings } = data  
+    const audienceName = audienceSettings?.audience_name ?? payload.segment_audience_key
+    
+    const audienceValue = payload.traits_or_props[payload.segment_audience_key]
+
+    let primaryIdentifier
+
+    switch(settings.identifier_type){
```
Comment (djluis) [inline packages/destination-actions/src/destinations/dynamic-yield-audiences/syncAudience/index.ts:110]: If I'm understanding correctly, this works for the primary identifiers, but later we need to add some dynamic code to fetch a custom identifier, like the example below:
```
default: {
   primaryIdentifier = payload[settings.identifier_type] ?? undefined
}
```

**Review by djluis — APPROVED** [review]
Looks good for the initial mapping with the primary identifiers, later we must include the custom "dynamic" identifiers. 👍 thanks again for the support @joe-ayoub-segment

File: packages/destination-actions/src/destinations/dynamic-yield-audiences/syncAudience/index.ts (hunk 1 above)
Comment (joe-ayoub-segment) [reply packages/destination-actions/src/destinations/dynamic-yield-audiences/syncAudience/index.ts:120]: > If I'm understanding correctly, this works for the primary identifiers, but later we need to add some dynamic code to fetch a custom identifier, like the example below:
> 
> ```
> default: {
>    primaryIdentifier = payload[settings.identifier_type] ?? undefined
> }
> ```

hi @djluis I decided to go ahead and hard code the 3 most common identifiers into a dropdown list, and to make email the default. If you are able to expose an API we can use to populate a dynamic list then we can make use of it.

Currently, the code only adds the identifier for the selected identifier_type into the payload which gets sent to DY. We could change this to include all identifiers we have, if you like. Though you'll have to show me how to structure that information in the payload.

File: packages/destination-actions/src/destinations/dynamic-yield-audiences/syncAudience/index.ts [hunk 2]
```diff
@@ -42,57 +76,90 @@ const action: ActionDefinition<Settings, Payload> = {
...
+            section_id : settings.sectionId,
+            identifier_type: settings.identifier_type,
+            accessKey: settings.accessKey
+          }
+        },
+        user_profiles : [ {  
+          user_identities : [ {   
+            type : settings.identifier_type,
+            encoding : "sha-256",
+            value : hashAndEncode(primaryIdentifier)
```
Comment (djluis) [inline packages/destination-actions/src/destinations/dynamic-yield-audiences/syncAudience/index.ts:155]: @joe-ayoub-segment We need to discuss this on next meeting ⚠️

**Review by djluis — CHANGES_REQUESTED** [review]

**Comment by joe-ayoub-segment** [comment]
PR deployed

**Pull request merged**